### ISSUE TITLE
resulted adoc raises metanorma gem error

### ISSUE THREAD

**Issue by Intelligent2013** [metanorma/sts2mn#52]
Resulted adoc:
```adoc
[arabic]
. text text term:[British Standard,British Standard];

. text text term:[intellectual property rights,intellectual property rights] set out in *<<sec_9.5.2>>*;

. test  the term:[committee] text term:[conflict of interest] (see *term:[conflict of interest]*);
+
--
NOTE: This does not apply in the case text text.
--
 
. text text 
 
. text text.

```

raises error:

```
         4: from C:/Ruby27-x64/lib/ruby/gems/2.7.0/gems/nokogiri-1.10.10-x64-mingw32/lib/nokogir/xml/node_set.rb:237:in `each'
         3: from C:/Ruby27-x64/lib/ruby/gems/2.7.0/gems/nokogiri-1.10.10-x64-mingw32/lib/nokogir/xml/node_set.rb:237:in `upto'
         2: from C:/Ruby27-x64/lib/ruby/gems/2.7.0/gems/nokogiri-1.10.10-x64-mingw32/lib/nokogir/xml/node_set.rb:238:in `block in each'
         1: from C:/Ruby27-x64/lib/ruby/gems/2.7.0/bundler/gems/metanorma-standoc-588211c242ee/lib/asciidocto/standoc/term_lookup_cleanup.rb:41:in `block in set_termxref_tags_target'
C:/Ruby27-x64/lib/ruby/gems/2.7.0/bundler/gems/metanorma-standoc-588211c242ee/lib/asciidoctor/standoc/term_lookup_cleanup.rb:55:in `remove_missing_ref': undefined method `content=' for nil:NilClass (NoMethodError)
```

**Comment by Intelligent2013** [comment]
Actually, problematic content in another:
```
[[sec_8.2]]
=== Title

Text text in the term:[,decision-making processes] (as defined in *term:[decision-making process]*) of a term:[technical committee,technical committee].

```
first value in `term:[,decision-making processes]` is empty.

**Comment by Intelligent2013** [comment]
Reason - issue in the source STS XML:
```
<named-content content-type="term" xlink:href="#">decision-making processes</named-content>
```
`xlink:href` is empty.
Solution - if value is empty, then replace it by named-content/text() (replace space, brackets, etc. to `-`).

**Comment by Intelligent2013** [comment]
One more error:
```
(XML Line 001295): Anchor introduction has already been used at line 113
	<clause id="introduction" inline-header="false" obligation="normative"><title>Introduction</title>
```
Reason - in source STS XML there are two Introduction sections:
```
<front>
  <sec id="sec_intro" sec-type="intro">
	<title>Introduction</title>
	<p>This document defines..
...
<back>
<app-group>
	<app content-type="normative-annex" id="sec_A">
		<label>Annex A</label>
		<annex-type>(normative)</annex-type>
		<title>Descriptions of the core-structure elements and attributes</title>
		<sec id="sec_A.1" sec-type="intro">
			<label>A.1</label>
			<title>Introduction</title>
			<p>This annex contains
```
Both sec have attribute `sec-type="intro"`, therefore in adocs:
00-02-intro.adoc
```adoc
[[introduction]]
== Introduction

This document defines a

```
annex-a.adoc:
```adoc
[[sec_A]]
[appendix,obligation=normative]
== Descriptions of the ...

[[introduction]]
=== Introduction

This annex contains
```

Solution - sec in `sec-type="intro"` in `front` or `body` sections should be marked as ``[[introduction]]``.

**Comment by Intelligent2013** [comment]
Source STS XML fragment: `<named-content content-type="ace-tag" specific-use="C2_start"/>` (start/end marker).
Resulted adoc: `term:[]`.
Solution: ignore `named-content` with `content-type="ace-tag"`  (temporary solution), relates to https://github.com/metanorma/mn-samples-bsi/issues/31.

**Comment by Intelligent2013** [comment]
Issues fixed.

**Comment by Intelligent2013** [comment]
Error:
```
         3: from C:/Ruby27-x64/lib/ruby/gems/2.7.0/bundler/gems/metanorma-bsi-eb0ba99b041a/lib/asciidoctor/bsi/validate.rb:88:in `section_style'
         2: from C:/Ruby27-x64/lib/ruby/gems/2.7.0/bundler/gems/metanorma-bsi-eb0ba99b041a/lib/asciidoctor/bsi/validate.rb:93:in `annex_validate'
         1: from C:/Ruby27-x64/lib/ruby/gems/2.7.0/bundler/gems/metanorma-bsi-eb0ba99b041a/lib/asciidoctor/bsi/validate.rb:108:in `annex_ordering_validate'
C:/Ruby27-x64/lib/ruby/gems/2.7.0/bundler/gems/metanorma-bsi-eb0ba99b041a/lib/asciidoctor/bsi/validate.rb:108:in `sort': comparison of String with String failed (ArgumentError)
```
occurs on the document pas\pas-2030\30400875\sections.1/annex-b.adoc[]

**Comment by Intelligent2013** [comment]
`bundle exec metanorma -t bsi -x xml sources/30400875/document.1.adoc` - ok
`bundle exec metanorma -t bsi -x presentation sources/30400875/document.1.adoc` - error

2.
sections.1/annex-a.adoc:
```adoc
[[sec_A]]
[appendix]
== Qualifications

*_COMMENTARY ON <<sec_A>>_*
...
```

sections.1/annex-b.adoc:
```adoc
[[sec_B]]
[appendix]
== Risk Assessments

[[sec_B.1]]
==== {blank}
The risk assessment process
...
```

document.1.adoc:
```
include::sections.1/annex-a.adoc[]

// include::sections.1/annex-b.adoc[]
```
generates xml ok:
```xml
<annex id="sec_A" inline-header="false" obligation="normative"><title><strong>Annex A</strong><tab/>(normative)<br/><strong>Qualifications</strong></title><p id="_d1daa063-2f25-46ca-8f10-5e097ad745fe">
<strong>
<em>COMMENTARY ON <xref target="sec_A">Annex A</xref></em>
</strong>
</p>
```

```
include::sections.1/annex-a.adoc[]

include::sections.1/annex-b.adoc[]
```
raises error:
```
C:/Ruby27-x64/lib/ruby/gems/2.7.0/bundler/gems/metanorma-bsi-eb0ba99b041a/lib/asciidoctor/bsi/validate.rb:108:in `sort': comparison of String with String failed (ArgumentError)
```

If remove sec_a from `*_COMMENTARY ON <<sec_A>>_*` -> `*_COMMENTARY ON_*`, then xml generates with both annex a and b.

**Comment by Intelligent2013** [comment]
Fixed in  metanorma-bsi 0.1.0.
